woocommerce/woocommerce-ios
# Auth Role Checks: Part 2.5 - Add accessibility support for role error screen

Refs #4478, #3963 
Depended by #4492, #4493

## Summary
Based on this [code review](https://github.com/woocommerce/woocommerce-ios/pull/4493#pullrequestreview-697643887) in #4493, there were problems with large text sizes. This PR provides accessibility support for large text. In a gist:

- The content view is nested in a scroll view, so the contents are vertically scrollable when using large text sizes.
- The link button does not have multi-line support by default, causing the UIButton to not respect the adjusted intrinsic content size. This PR addresses that.
- Since the action buttons are vertically stacked _and_ the secondary button title is rather long, I've applied a maximum font size for the buttons to allow for readable space for the content. Note that **this is only applied in iPhone interface idioms**.
- The `class` to `AnyObject` change in `RoleErrorOutput` is to get rid of an Xcode warning.

## 👀 Screenshots

• | Portrait | Landscape
--|--|--
iPhone, normal size | ![4478_a11y_reg_iphone_portrait](https://user-images.githubusercontent.com/1299411/125661237-1fd8054c-7c22-4999-b45d-5a1496ab803f.png) | ![4478_a11y_reg_iphone_landscape](https://user-images.githubusercontent.com/1299411/125661235-ac1061b0-6018-41cd-9f5e-2693c885ae42.png)
iPhone, large size | ![4478_a11y_xl_iphone_portrait](https://user-images.githubusercontent.com/1299411/125661244-df792030-b1e3-4562-abbd-d7ce7cd81f06.png) | ![4478_a11y_xl_iphone_landscape](https://user-images.githubusercontent.com/1299411/125661241-0c4d190e-3b5b-49c2-b73d-20fb2d8af1fd.png)
iPad, normal size | ![4478_a11y_reg_ipad_portrait](https://user-images.githubusercontent.com/1299411/125661230-c8032d66-d3ee-42db-81b7-47f938dc2ba6.png) | ![4478_a11y_reg_ipad_landscape](https://user-images.githubusercontent.com/1299411/125661226-0aad6e19-5100-415c-bf58-3c8d202b6ae9.png)
iPad, large size | ![4478_a11y_ipad_portrait](https://user-images.githubusercontent.com/1299411/125661218-582d755b-2ab7-4d49-9b3e-4d37e45eb1a2.png) | ![4478_a11y_ipad_landscape](https://user-images.githubusercontent.com/1299411/125661200-22c4cd7a-160a-42bf-9eaa-867b4f8b3a61.png)

Note: for dark mode looks, please refer to #4478.

## To Test
Currently, there's no path present to reach the new role error page. For the sake of just testing the layout, you'd have to intercept with the navigation flow temporarily. Modify `AppCoordinator` like so:

```swift
// AppCoordinator.swift:L38
guard let self = self else { return }

// add these lines to always show the role error page.
let viewModel = RoleErrorViewModel(displayName: "John Doe", roles: ["editor"])
let viewController = RoleErrorViewController(viewModel: viewModel)
let navigationController = WooNavigationController(rootViewController: viewController)
self.setWindowRootViewControllerAndAnimateIfNeeded(navigationController)
self.isLoggedIn = isLoggedIn
return
```
When running the app again, you should see the role error page pop up first thing.

## Author Checklist
Update release notes:
